src/ctapipe/irf/select.py

```diff
@@ -0,0 +1,298 @@
+            if c not in rename_to:
+                raise ValueError(
+                    f"No column corresponding to {c} is defined in "
+                    f"EventPreProcessor.rename_columns and {standard_renames[c]} "
+                    "is not in the given data."
+                )
+
+        keep_columns.extend(rename_from)
+        events = QTable(events[keep_columns], copy=False)
+        events.rename_columns(rename_from, rename_to)
+        return events
+
+    def make_empty_table(self) -> QTable:
+        """
+        This function defines the columns later functions expect to be present
+        in the event table.
+        """
+        columns = [
+            "obs_id",
+            "event_id",
+            "true_energy",
+            "true_az",
+            "true_alt",
+            "reco_energy",
+            "reco_az",
+            "reco_alt",
+            "reco_fov_lat",
+            "reco_fov_lon",
+            "gh_score",
+            "pointing_az",
+            "pointing_alt",
+            "theta",
+            "true_source_fov_offset",
+            "reco_source_fov_offset",
+        ]
+        units = {
+            "true_energy": u.TeV,
+            "true_az": u.deg,
+            "true_alt": u.deg,
+            "reco_energy": u.TeV,
+            "reco_az": u.deg,
+            "reco_alt": u.deg,
+            "reco_fov_lat": u.deg,
+            "reco_fov_lon": u.deg,
+            "pointing_az": u.deg,
+            "pointing_alt": u.deg,
+            "theta": u.deg,
+            "true_source_fov_offset": u.deg,
+            "reco_source_fov_offset": u.deg,
+        }
+        descriptions = {
+            "obs_id": "Observation Block ID",
+            "event_id": "Array Event ID",
+            "true_energy": "Simulated Energy",
+            "true_az": "Simulated azimuth",
+            "true_alt": "Simulated altitude",
+            "reco_energy": "Reconstructed energy",
+            "reco_az": "Reconstructed azimuth",
+            "reco_alt": "Reconstructed altitude",
+            "reco_fov_lat": "Reconstructed field of view lat",
+            "reco_fov_lon": "Reconstructed field of view lon",
+            "pointing_az": "Pointing azimuth",
+            "pointing_alt": "Pointing altitude",
+            "theta": "Reconstructed angular offset from source position",
+            "true_source_fov_offset": "Simulated angular offset from pointing direction",
+            "reco_source_fov_offset": "Reconstructed angular offset from pointing direction",
+            "gh_score": "prediction of the classifier, defined between [0,1], where values close to 1 mean that the positive class (e.g. gamma in gamma-ray analysis) is more likely",
+        }
+
+        return QTable(names=columns, units=units, descriptions=descriptions)
+
+
+class EventsLoader(Component):
+    classes = [EventPreProcessor]
+
+    def __init__(self, kind: str, file: Path, target_spectrum: Spectra, **kwargs):
```

Comment:
why are `kind`, `file` and `target_spectrum` keyword arguments here and not done via the configuration system?

Reply:
Since multiple files are loaded when running the tools, these configuration options would have to be overwritten within the tool. This is not intended is it?